subprojects/core/src/main/java/org/gradle/testfixtures/internal/ProjectBuilderImpl.java

```diff
@@ -107,6 +107,8 @@ public ProjectInternal createProject(String name, File inputProjectDir, File gra
         File userHomeDir = gradleUserHomeDir == null ? new File(projectDir, "userHome") : FileUtils.canonicalize(gradleUserHomeDir);
         StartParameterInternal startParameter = new StartParameterInternal();
         startParameter.setGradleUserHomeDir(userHomeDir);
+        startParameter.useEmptySettings();
+        startParameter.setCurrentDir(inputProjectDir);
```

Comment:
Why are these changes also needed as a part of this larger change?

Reply:
The checkstyle plugin uses the FileFactory service. Without these changes the invoked service in the Project mockup reports the wrong folder and not the temp folder project root as expected